src/app/pages/questions/containers/questions-page.component.ts

```diff
@@ -213,15 +215,16 @@ export class QuestionsPageComponent {
     let increment = 1
     let questionIdx = this.currentQuestion + 1
     const questionFieldName = this.questions[this.currentQuestion].field_name
-    const responses = Object.assign({}, this.answerService.answers)
-    const answer = responses[questionFieldName]
-    let answerLength = answer.length
     if (questionIdx < this.questions.length) {
       while (this.questions[questionIdx].evaluated_logic !== '') {
+        const responses = Object.assign({}, this.answerService.answers)
         const logic = this.questions[questionIdx].evaluated_logic
+        const logicFieldName = this.getLogicFieldName(logic)
+        const answer = this.answerService.answers[logicFieldName]
+        let answerLength = answer.length
         if (answerLength) {
           while (answerLength > 0) {
-            responses[questionFieldName] = answer[answerLength - 1]
+            responses[logicFieldName] = answer[answerLength - 1]
```

Comment:
You can use a for loop here -

```javascript
for(let answerLength = answer.length; answerLength > 0; answerLength--) {
     responses[logicFieldName] = answer[answerLength - 1]
     if (eval(logic) === true) return increment
}
```

Also any particular reason for using the loop backwards??

Reply:
Also how will this work if we have more than 2 responses to eval together? 
For example,
```json
 "responses['esm_social_interact'] == 1 && responses['esm_social_interact'] == 2
```
i.e. if we have to evaluate `&&` too ?

Reply:
Tracking this via #228